source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -0,0 +1,308 @@
+#include "extensions/filters/http/common/compressor/compressor.h"
+
+#include "common/http/header_map_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace Compressors {
+
+namespace {
+
+// Default minimum length of an upstream response that allows compression.
+const uint64_t DefaultMinimumContentLength = 30;
+
+// Default content types will be used if any is provided by the user.
+const std::vector<std::string>& defaultContentEncoding() {
+  CONSTRUCT_ON_FIRST_USE(std::vector<std::string>,
+                         {"text/html", "text/plain", "text/css", "application/javascript",
+                          "application/json", "image/svg+xml", "text/xml",
+                          "application/xhtml+xml"});
+}
+
+} // namespace
+
+std::vector<std::string> CompressorFilterConfig::registered_compressors_ = {};
```

Comment:
no static non-pods.  See https://github.com/envoyproxy/envoy/blob/master/STYLE.md , search for 'fiasco'. You can easily work around with a lazy-init.